# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaRecorder.cpp
# Commit: 3f2015f34fc2
# Full Hash: 3f2015f34fc2daaaff9d90b7114157face56edf2
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2022-09-14 04:09:22
# Description:
#   Bug 1713940, null check owner before using it, r=padenot
#   
#   Reused an existing wpt. Added also some .step() calls there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D157143
# ==============================================================================

diff -r dff049971660 -r 3f2015f34fc2 dom/media/MediaRecorder.cpp
--- a/dom/media/MediaRecorder.cpp	Tue Sep 13 17:08:46 2022 +0000
+++ b/dom/media/MediaRecorder.cpp	Tue Sep 13 17:09:24 2022 +0000
@@ -1242,6 +1242,12 @@
   //    throw a SecurityError DOMException and abort these steps.
   if (mStream) {
     RefPtr<nsIPrincipal> streamPrincipal = mStream->GetPrincipal();
+    if (!streamPrincipal) {
+      // This is more or less part of the step 7, see below.
+      aResult.ThrowNotSupportedError("The MediaStream is inactive");
+      return;
+    }
+
     if (!PrincipalSubsumes(this, streamPrincipal)) {
       aResult.ThrowSecurityError(
           "The MediaStream's isolation properties disallow access from "